runtime/vm/jvminit.c

```diff
@@ -2557,6 +2557,11 @@ VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved)
 				parseErrorOption = VMOPT_XXMAXDIRECTMEMORYSIZEEQUALS;
 				goto _memParseError;
 			}
+			if ((vm->directByteBufferMemoryMax != ~(UDATA)0) && (vm->directByteBufferMemoryMax > ((1 << 63) - 1))) {
```

Comment:
I don't think this works for 32-bit.

Reply:
You're right... I'm in the process of reworking this to be more platform/width independent.